Add new snapshot to dcp3 on HCA dev

The new Project ID 6307bc26-6276-4b85-bd26-c403d52b6eb2 needs to be added to the HCA dev dcp3 catalog.

The new project is listed as row 131 in the "HCA Dev" tab of the [TDR Datasets and snapshots](https://docs.google.com/spreadsheets/d/1yYiB3TrX0wvLCHvtWY1GTZZcAf5MmMREgn52OKCHsqs/edit?usp=sharing) spreadsheet.

([Slack thread](https://humancellatlas.slack.com/archives/C01360XN04S/p1762980640293809?thread_ts=1761840570.029599&cid=C01360XN04S))

Only one notification failed to index in my personal deployment due to an invalid schema:

`[WARNING]	2025-11-14T09:22:16.045Z	57f9fbce-e947-5e44-b4d3-9a5e1e42f2e1	azul.indexer.action_controller	Worker failed to handle message SQSMessage(body={'action': 'add', 'catalog': 'dcp3', 'bundle_fqid': {'uuid': 'cd1f3250-bba8-5797-acb8-d8225c98575a', 'version': '2025-10-17T14:49:39.930000Z', 'source': {'id': '0a4bd5cf-f007-49b2-98e7-f328490ede47', 'spec': 'tdr:bigquery:gcp:datarepo-dev-8a52f620:hca_dev_6307bc2662764b85bd26c403d52b6eb2__20251110_dcp2_20251110_dcp99', 'prefix': '/0', 'type': 'azul.terra.TDRSourceRef'}}, 'bundle_partition': {'prefix_length': 0, 'prefix': 0}}, attempts=1, id='287b2dca-885b-46d5-b922-4022c90c4290')`
```
Traceback (most recent call last):
  File ""/var/task/azul/indexer/action_controller.py"", line 80, in _handle_events
    message_handler(action, message.body)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File ""/var/task/azul/indexer/index_queue_service.py"", line 193, in contribute
    contributions, replicas = self.transform(catalog,
                              ~~~~~~~~~~~~~~^^^^^^^^^
                                             bundle_fqid,
                                             ^^^^^^^^^^^^
                                             bundle_partition,
                                             ^^^^^^^^^^^^^^^^^
                                             delete=delete)
                                             ^^^^^^^^^^^^^^
  File ""/var/task/azul/indexer/index_queue_service.py"", line 234, in transform
    results = self.index_service.transform(catalog,
                                           bundle,
                                           bundle_partition,
                                           delete=delete)
  File ""/var/task/azul/indexer/index_service.py"", line 273, in transform
    transformers = plugin.transformers(bundle, delete=delete)
  File ""/var/task/azul/plugins/metadata/hca/__init__.py"", line 100, in transformers
    api_bundle = api.Bundle(uuid=bundle.uuid,
                            version=bundle.version,
    ...<2 lines>...
                            links_json=bundle.links,
                            stitched_entity_ids=bundle.stitched)
  File ""/var/task/humancellatlas/data/metadata/api.py"", line 967, in __init__
    self.projects = from_json_vx(Project)
                    ~~~~~~~~~~~~^^^^^^^^^
  File ""/var/task/humancellatlas/data/metadata/api.py"", line 965, in from_json_vx
    return {entity.document_id: entity for entity in entities}
                                                     ^^^^^^^^
  File ""/var/task/humancellatlas/data/metadata/api.py"", line 962, in <genexpr>
    core_cls.from_json(**args)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^
  File ""/var/task/humancellatlas/data/metadata/api.py"", line 120, in from_json
    described_by = cls.validate_described_by(content)
  File ""/var/task/humancellatlas/data/metadata/api.py"", line 131, in validate_described_by
    assert described_by.netloc in valid_schema_domains, R(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: R('Unexpected schema domain', furl('https://schema.staging.data.humancellatlas.org/type/project/16.0.0/project'), ['schema.humancellatlas.org'])
```

--- a/UPGRADING.rst
+++ b/UPGRADING.rst
@@ -20,6 +20,24 @@ reverted. This is all fairly informal and loosely defined. Hopefully we won't
 have too many entries in this file.
 
 
+#7571 Add new snapshot to dcp3 on HCA dev
+=========================================
+
+Operator
+--------
+
+Perform a targeted reindex of snapshot
+``datarepo-dev-b46086a9.hca_dev_51f02950ee254f4b8d0759aa99bb3498__20210827_20230215``
+is ``sandbox`` and ``dev``.
+
+Everyone
+--------
+
+In your HCA personal deployments' ``environment.py`` files, add the new snapshot
+to the sources for catalog ``dcp3``, bringing the total number of sources to
+105. Then perform a targeted reindex of that snapshot.
+
+
 #7066 Configure mirroring per catalog and source
 ================================================
 

--- a/deployments/dev/environment.py
+++ b/deployments/dev/environment.py
@@ -168,15 +168,18 @@ dcp2_sources = union({}, 105, delta([
     source('bigquery', 'datarepo-dev-96d8e08c', 'hca_dev_faeedcb0e0464be7b1ad80a3eeabb066__20210831_20210903'),
 ]))
 
-dcp3_sources = union(dcp2_sources, 104, delta([
+dcp3_sources = union(dcp2_sources, 105, delta([
+    # @formatter:off
     source('bigquery', 'datarepo-dev-3d6d24ad', 'hca_dev_05657a599f9d4bb9b77b24be13aa5cea__20210827_20230215'),
     source('bigquery', 'datarepo-dev-76de829d', 'hca_dev_2043c65a1cf84828a6569e247d4e64f1__20210831_20210907', pop),
     source('bigquery', 'datarepo-dev-b46086a9', 'hca_dev_51f02950ee254f4b8d0759aa99bb3498__20210827_20230215'),
+    source('bigquery', 'datarepo-dev-8a52f620', 'hca_dev_6307bc2662764b85bd26c403d52b6eb2__20251110_dcp2_20251110_dcp99'),  # noqa E501
     source('bigquery', 'datarepo-dev-27c50fbc', 'hca_dev_c5f4661568de4cf4bbc2a0ae10f08243__20210827_20230215'),
     source('bigquery', 'datarepo-dev-2a4ab485', 'hca_dev_d2111fac3fc44f429b6d32cd6a828267__20210830_20230215'),
     source('bigquery', 'datarepo-dev-1005632d', 'hca_dev_d3ac7c1b53024804b611dad9f89c049d__20210827_20230215'),
     source('bigquery', 'datarepo-dev-8709b362', 'hca_dev_df88f39f01a84b5b92f43177d6c0f242__20210827_20230215'),
     source('bigquery', 'datarepo-dev-e1712bfa', 'hca_dev_e526d91dcf3a44cb80c5fd7676b55a1d__20210902_20230215'),
+    # @formatter:on
 ]))
 
 lungmap_sources = union({}, 2, delta([

--- a/deployments/sandbox/environment.py
+++ b/deployments/sandbox/environment.py
@@ -170,15 +170,18 @@ dcp2_sources = union({}, 105, delta([
     source('bigquery', 'datarepo-dev-96d8e08c', 'hca_dev_faeedcb0e0464be7b1ad80a3eeabb066__20210831_20210903'),
 ]))
 
-dcp3_sources = union(dcp2_sources, 104, delta([
+dcp3_sources = union(dcp2_sources, 105, delta([
+    # @formatter:off
     source('bigquery', 'datarepo-dev-3d6d24ad', 'hca_dev_05657a599f9d4bb9b77b24be13aa5cea__20210827_20230215'),
     source('bigquery', 'datarepo-dev-76de829d', 'hca_dev_2043c65a1cf84828a6569e247d4e64f1__20210831_20210907', pop),
     source('bigquery', 'datarepo-dev-b46086a9', 'hca_dev_51f02950ee254f4b8d0759aa99bb3498__20210827_20230215'),
+    source('bigquery', 'datarepo-dev-8a52f620', 'hca_dev_6307bc2662764b85bd26c403d52b6eb2__20251110_dcp2_20251110_dcp99'),  # noqa E501
     source('bigquery', 'datarepo-dev-27c50fbc', 'hca_dev_c5f4661568de4cf4bbc2a0ae10f08243__20210827_20230215'),
     source('bigquery', 'datarepo-dev-2a4ab485', 'hca_dev_d2111fac3fc44f429b6d32cd6a828267__20210830_20230215'),
     source('bigquery', 'datarepo-dev-1005632d', 'hca_dev_d3ac7c1b53024804b611dad9f89c049d__20210827_20230215'),
     source('bigquery', 'datarepo-dev-8709b362', 'hca_dev_df88f39f01a84b5b92f43177d6c0f242__20210827_20230215'),
     source('bigquery', 'datarepo-dev-e1712bfa', 'hca_dev_e526d91dcf3a44cb80c5fd7676b55a1d__20210902_20230215'),
+    # @formatter:on
 ]))
 
 lungmap_sources = union({}, 2, delta([

--- a/src/humancellatlas/data/metadata/api.py
+++ b/src/humancellatlas/data/metadata/api.py
@@ -37,6 +37,7 @@ from furl import (
 from azul import (
     R,
     cached_property,
+    config,
 )
 from azul.collections import (
     LookupDefault,
@@ -102,6 +103,8 @@ class ManifestEntry:
 
 
 valid_schema_domains = ['schema.humancellatlas.org']
+if not config.deployment.is_stable:
+    valid_schema_domains.append('schema.staging.data.humancellatlas.org')
 
 
 @dataclass(init=False)

